dataline-workers/src/test/java/io/dataline/workers/DefaultSyncWorkerTest.java

```diff
@@ -0,0 +1,171 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.dataline.workers;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.dataline.commons.functional.CloseableConsumer;
+import io.dataline.config.Column;
+import io.dataline.config.DataType;
+import io.dataline.config.DestinationConnectionImplementation;
+import io.dataline.config.Schema;
+import io.dataline.config.SingerMessage;
+import io.dataline.config.SourceConnectionImplementation;
+import io.dataline.config.StandardSync;
+import io.dataline.config.StandardSyncInput;
+import io.dataline.config.StandardSyncSummary;
+import io.dataline.config.StandardTapConfig;
+import io.dataline.config.StandardTargetConfig;
+import io.dataline.config.State;
+import io.dataline.config.Table;
+import io.dataline.workers.protocol.singer.MessageFactory;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.UUID;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+
+class DefaultSyncWorkerTest extends BaseWorkerTestCase {
+  private static final Path WORKSPACE_ROOT = Path.of("/workspaces/10");
+  private static final String TABLE_NAME = "user_preferences";
+  private static final String COLUMN_NAME = "favorite_color";
+  private static final long LAST_SYNC_TIME = 1598565106;
+  private static final UUID WORKSPACE_ID = UUID.randomUUID();
+  private static final UUID SOURCE_SPECIFICATION_ID = UUID.randomUUID();
+  private static final UUID SOURCE_IMPLEMENTATION_ID = UUID.randomUUID();
+  private static final UUID DESTINATION_SPECIFICATION_ID = UUID.randomUUID();
+  private static final UUID DESTINATION_IMPLEMENTATION_ID = UUID.randomUUID();
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void test() throws Exception {
+    final ObjectMapper objectMapper = new ObjectMapper();
+
+    final ObjectNode sourceConnection = objectMapper.createObjectNode();
+    sourceConnection.put("apiKey", "123");
+    sourceConnection.put("region", "us-east");
+
+    final ObjectNode destinationConnection = objectMapper.createObjectNode();
+    destinationConnection.put("username", "dataline");
+    destinationConnection.put("token", "anau81b");
+
+    final SourceConnectionImplementation sourceConnectionConfig =
+        new SourceConnectionImplementation();
+    sourceConnectionConfig.setConfiguration(sourceConnection);
```

Comment:
is the configuration field always going to be a json object?

Reply:
yes.